ledger/accountdb.go

```diff
@@ -844,34 +858,79 @@ func accountsPutTotals(tx *sql.Tx, totals AccountTotals, catchpointStaging bool)
 	return err
 }
 
-// accountsNewRound updates the accountbase and assetcreators by applying the provided deltas to the accounts / creatables.
-func accountsNewRound(tx *sql.Tx, updates map[basics.Address]accountDeltaCount, creatables map[basics.CreatableIndex]modifiedCreatable, proto config.ConsensusParams) (err error) {
+// accountsNewRound updates the accountbase and assetcreators tables by applying the provided deltas to the accounts / creatables.
+// The updates.old entries are being updated to reflect the latest value that was written to the database.
+func accountsNewRound(tx *sql.Tx, updates map[basics.Address]accountDeltaCount, creatables map[basics.CreatableIndex]modifiedCreatable, proto config.ConsensusParams, lastUpdateRound basics.Round) (err error) {
```

Comment:
imo updating `updates` is very tricky. I understand why it is done (optimization) but maybe return `updates` as well? It will be a shallow copy. Or a full copy since `updates` is completely updated.

Reply:
While I don't really see any value in returning the original input/output parameter, I'm ok doing so if you think it would make the code cleaner. Note that the same "updates" variable that is provided here as the input is also being used in the `commitRound` loop for updating the ndeltas, which makes it the ideal place to store the updated result, since the map is already been iterated upon.